main/wifi.c

```diff
@@ -77,14 +78,63 @@ static esp_err_t event_handler(void *ctx, system_event_t *event)
     return ESP_OK;
 }
 
-int wifi_connect(const char *ssid, const char *password)
+eap_method_t wifi_eap_atomethod(const char *method)
 {
+    if (method == NULL)
+        return EAP_NONE;
+
+    struct {
+        const char *name;
+        int method;
+    } *p, methods[] = {
+        { "TLS", EAP_TLS },
+        { "PEAP", EAP_PEAP },
+        { "TTLS", EAP_TTLS },
+        { NULL, EAP_NONE }
+    };
+
+    for (p = methods; p->name; p++)
+    {
+        if (!strcmp(p->name, method))
+            break;
+    }
+
+    return p->method;
+}
+
+int wifi_connect(const char *ssid, const char *password,
+    eap_method_t eap_method, const char *eap_identity,
+    const char *eap_username, const char *eap_password,
+    const char *ca_cert, const char *client_cert, const char *client_key)
+{
+    esp_wpa2_config_t config = WPA2_CONFIG_INIT_DEFAULT();
     wifi_config_t wifi_config = {};
     strncpy((char *)wifi_config.sta.ssid, ssid, 32);
-    strncpy((char *)wifi_config.sta.password, password, 64);
+    if (password)
+        strncpy((char *)wifi_config.sta.password, password, 64);
 
     ESP_ERROR_CHECK(esp_wifi_set_mode(WIFI_MODE_STA));
     ESP_ERROR_CHECK(esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config));
+    if (eap_method)
+    {
+        ESP_ERROR_CHECK(esp_wifi_sta_wpa2_ent_set_ca_cert((uint8_t *)ca_cert,
```

Comment:
Please check for `ca_cert` before calling `esp_wifi_sta_wpa2_ent_set_ca_cert()`.

Reply:
Sorry about that!